File: quarb-code/tests/fixtures/vocab.py
import os
from sys import path


class Lexer:
    """A lexer."""

    def __init__(self, text):
        self.text = text

    def lex(self):
        """Scan the input."""
        out = []
        for c in self.text:
            if c.isalnum():
                out.append(c)
            elif c == " ":
                out.append("_")
            else:
                out.append(".")
        return out


def helper(a, b):
    f = lambda x: x + 1
    while a < b:
        a = f(a)
    match a:
        case 0:
            return 0
        case _:
            return a


def run(n):
    return helper(n, 2)

File: quarb-code/tests/fixtures/vocab.js
import fs from "fs";

/** A lexer. */
class Lexer {
  size = 0;

  constructor(text) {
    this.text = text;
  }

  /** Scan the input. */
  lex() {
    const out = [];
    for (const c of this.text) {
      if (c === " ") {
        out.push("_");
      } else {
        out.push(c);
      }
    }
    return out;
  }

  get length() {
    return this.size;
  }
}

const helper = (a, b) => {
  let i = 0;
  while (i < a) {
    i += 1;
  }
  switch (a) {
    case 0:
      return b;
    default:
      return new Lexer("x").size;
  }
};

function main() {
  const l = new Lexer("hi");
  do {
    l.size += 1;
  } while (l.size < 2);
  return helper(l.size, 2);
}
